### PR TITLE
Security Alerts Workbook

### PR DESCRIPTION
**Pull request by dicolanl**
Fixes #

## Proposed Changes
New Azure Sentinel Security Alerts workbook
  -
  -
  -

### CODE REVIEW

File: Workbooks/WorkbooksMetadata.json [hunk 1]
```diff
@@ -479,5 +479,18 @@
     "templateRelativePath": "OneIdentity.json",
     "subtitle": "",
     "provider": "One Identity LLC.​"
-  }
+  },
+  {
+    "workbookKey": "AzureSentinelSecurityAlerts",
```
Comment (Amitbergman) [inline Workbooks/WorkbooksMetadata.json]: Our convention is to add the word "Workbook" in the end of the workbook key.
Look at the other workbook keys.

File: Workbooks/WorkbooksMetadata.json [hunk 2]
```diff
@@ -479,5 +479,18 @@
     "templateRelativePath": "OneIdentity.json",
     "subtitle": "",
     "provider": "One Identity LLC.​"
-  }
+  },
+  {
+    "workbookKey": "AzureSentinelSecurityAlerts",
+    "logoFileName": "Azure_Sentinel.svg",
+    "description": "Security Alerts dashboard for alerts in your Azure Sentinel environment.",
+    "dataTypesDependencies": [ "SecurityAlerts" ],
```
Comment (Amitbergman) [inline Workbooks/WorkbooksMetadata.json]: I think the relevant data type is "SecurityAlert" and not "SecurityAlert**s**", right?

File: Workbooks/AzureSentinelSecurityAlerts.json [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
...r\n| extend Target = iif(Entity_Type == \"account\", Entity_Name, iif(Entity_Type == \"azure-resource\", Entity_ResourceId, iif(Entity_Type == \"cloud-application\", Entity_Name, iif(Entity_Type == \"dns\", Entity_Name, iif(Entity_Type == \"file\", strcat(Entity_Directory, \"\\\\\", Entity_Name), iif(Entity_Type == \"filehash\", Entity_Value, iif(Entity_Type == \"host\", Entity_HostName, iif(Entity_Type == \"ip\" , Entity_Address, iif(Entity_Type == \"malware\", Entity_HostName, iif(Entity_Type == \"network-connection\", Entity_Name, iif(Entity_Type == \"process\", Entity_ProcessId, iif(Entity_Type == \"registry-key\", Entity_Name, iif(Entity_Type == \"registry-value\", Entity_Name, iif(Entity_Type == \"security-group\", Entity_Name, iif(Entity_Type == \"url\", Entity_Url, \"NoTarget\")))))))))))))))\r\n| where Entity_Type in (\"account\", \"host\", \"ip\", \"url\", \"azure-resource\", \"cloud-application\", \"dns\", \"file\", \"filehash\", \"malware\", \"network-connection\", \"process\", \"registry-key\", \"registry-value\", \"security-group\")\r\n| summarize count() by bin(TimeGenerated, 1d), Target, Entity_Type\r\n| project-away TimeGenerated\r\n| order by count_ desc\r\n\r\n",
```
Comment (Amitbergman) [inline Workbooks/AzureSentinelSecurityAlerts.json]: When I try to see this workbook in my workspace, it gives me an error on this query:
SecurityAlert
| where "{AlertSeverity:lable}" == "All" or AlertSeverity in ({AlertSeverity})
| where "{ProductName:lable}" == "All" or ProductName in ({ProductName})
| where AlertSeverity == '{AlertSeverityPicker}' or '{AlertSeverityPicker}' == "All"
| where ProductName == '{ProductNamePicker}' or '{ProductNamePicker}' == "All"
| extend Entities = iff(isempty(Entities), todynamic('[{"dummy" : ""}]'), todynamic(Entities))
| mvexpand Entities
| evaluate bag_unpack(Entities, "Entity_")
| extend Entity_Type = columnifexists("Entity_Type", "")
| extend Target = iif(Entity_Type == "account", Entity_Name, iif(Entity_Type == "azure-resource", Entity_ResourceId, iif(Entity_Type == "cloud-application", Entity_Name, iif(Entity_Type == "dns", Entity_Name, iif(Entity_Type == "file", strcat(Entity_Directory, "\\", Entity_Name), iif(Entity_Type == "filehash", Entity_Value, iif(Entity_Type == "host", Entity_HostName, iif(Entity_Type == "ip" , Entity_Address, iif(Entity_Type == "malware", Entity_HostName, iif(Entity_Type == "network-connection", Entity_Name, iif(Entity_Type == "process", Entity_ProcessId, iif(Entity_Type == "registry-key", Entity_Name, iif(Entity_Type == "registry-value", Entity_Name, iif(Entity_Type == "security-group", Entity_Name, iif(Entity_Type == "url", Entity_Url, "NoTarget")))))))))))))))
| where Entity_Type in ("account", "host", "ip", "url", "azure-resource", "cloud-application", "dns", "file", "filehash", "malware", "network-connection", "process", "registry-key", "registry-value", "security-group")
| summarize count() by bin(TimeGenerated, 1d), Target, Entity_Type
| project-away TimeGenerated
| order by count_ desc

It tells me that "Extend_Uri" does not exist.
Does it also happen to you?

image

Comment (dicolanl) [reply Workbooks/AzureSentinelSecurityAlerts.json]: image

works fine for me.

**Pull request merged**